packages/react-native/ReactCommon/react/nativemodule/microtasks/IdleDeadline.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+* Copyright (c) Meta Platforms, Inc. and affiliates.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+#include <react/renderer/runtimescheduler/RuntimeSchedulerBinding.h>
+#include <react/renderer/runtimescheduler/RuntimeScheduler.h>
+#include "IdleDeadline.h"
+#include <chrono>
+
+namespace facebook::react {
+
+IdleDeadline::IdleDeadline(
+    bool didTimeout,
+    RuntimeSchedulerTimePoint deadline)
+    : didTimeout_(didTimeout), deadline_(deadline) {}
+
+jsi::Value IdleDeadline::get(
+    jsi::Runtime& runtime,
+    const jsi::PropNameID& name) {
+  using namespace std::chrono_literals;
+
+  auto propertyName = name.utf8(runtime);
+
+  if (propertyName == "didTimeout") {
+    return jsi::Value(runtime, didTimeout_);
+  }
+
+  if (propertyName == "timeRemaining") {
+    auto binding = RuntimeSchedulerBinding::getBinding(runtime);
+    auto scheduler = binding->getScheduler();
+
+    return jsi::Function::createFromHostFunction(
+        runtime,
+        name,
+        0,
+        [&, this](
+            jsi::Runtime& runtime,
+            const jsi::Value&,
+            const jsi::Value*,
+            size_t) noexcept -> jsi::Value {
+          auto now = scheduler->now();
```

Comment:
Before this, we should check if `scheduler->getShouldYield()` is true. If it is, we should return 0 here before checking anything else.